src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,325 @@
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="gpioController">The GPIO controller used for communication.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, IGpioController gpioController, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize)
+        {
+            if (!InRange((uint)spiBufferSize, 0x1000, 0x10000))
+            {
+                throw new ArgumentException($"SPI Buffer Size must be between 4096 and 65536.", nameof(spiBufferSize));
+            }
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+            _dcPinId = dataCommandPin;
+            _resetPinId = resetPin;
+
+            _gpioDevice = gpioController;
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            _spiBufferSize = spiBufferSize;
+
+            SendCommand(new SetSegmentReMapColorDepth());
+        }
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize) : this(spiDevice, new GpioController(), dataCommandPin, resetPin, spiBufferSize)
+        {
+            //as a default GPIO controller has been used then dispose of it later..
+            _disposeGpioController = true;
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple represening the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
+        private (byte, byte) Color565(Color color)
```

Comment:
You don't have to do it now, but it bight be worth it to move this method in common as a helper, since I know that several bindings would benefit from this. I know that at least the SenseHat binding is also doing this.

Reply:
@joperezr 

Well... I have not done this as the Ssd1351 device can either use RGB565 or BGR565 and so this function is more complex than the one used in the sense hat. Perhaps this should be done when we have more display examples.